### ISSUE TITLE
Stash aliases, xhv_namecount/mult. enames

### ISSUE THREAD

**Issue by toddr** [rurban/perl-compiler#331]
stash aliases fail with all versions, 5.6 - 5.22
```
package Count;
sub getline { print "ok\n" };
BEGIN { *The::Count:: = \*Count::; } ;
The::Count->getline();
```

perlcc
```
$>perlcc -r -e 'package Count; sub getline { print "ok\n" }; BEGIN { *The::Count:: = \*Count::; } ; The::Count->getline'
Can't locate object method "getline" via package "The::Count" (perhaps you forgot to load "The::Count"?) at -e line 1.
```

**Comment by toddr** [comment]
Switching the BEGIN block to INIT fixes things.
```
perlcc -r -e 'package Count; sub getline { print "ok\n" }; INIT { *The::Count:: = \*Count::; } ; The::Count->getline'
ok
```

**Comment by rurban** [comment]
We are missing to store all stash names, if namecount > 1

perl -e 'use Devel::Peek; package Count;sub getline{print "ok\n" };BEGIN{*The::Count::=\*Count::;}; Devel::Peek::Dump(\%The::Count::);The::Count->getline();'
```
  SV = PVHV(0x1a3c1d0) at 0x1a35ed8
    REFCNT = 2
    FLAGS = (OOK,SHAREKEYS)
    ARRAY = 0x1a7b4b0  (0:6, 1:2)
    hash quality = 125.0%
    KEYS = 2
    FILL = 2
    MAX = 7
    RITER = -1
    EITER = 0x0
    NAME = "Count"
    NAMECOUNT = 2
    ENAME = "Count", "The::Count"
    BACKREFS = 0x1a54040
```
pcc =>
```
  SV = PVHV(0x220ae50) at 0x2217b40
    REFCNT = 2
    FLAGS = (OOK,SHAREKEYS)
    ARRAY = 0x221cc00  (0:5, 1:3)
    hash quality = 150.0%
    KEYS = 3
    FILL = 3
    MAX = 7
    RITER = -1
    EITER = 0x0
    NAME = "Count"
    ENAME = "Count"
    BACKREFS = 0x2240200
```

**Comment by rurban** [comment]
5.14 coretests:
```
t/C-COMPILED/CORE--io--layers==BADTEST-203.t           (Wstat: 0 Tests: 23 Failed: 0)
  TODO passed:   12-13, 22-23
t/C-COMPILED/CORE--io--utf8==BADTEST-319.t             (Wstat: 0 Tests: 23 Failed: 0)
  TODO passed:   12-13, 22-23
t/C-COMPILED/CORE--mro--isarev==BADTEST-353.t          (Wstat: 0 Tests: 23 Failed: 0)
  TODO passed:   12-13, 22-23
t/C-COMPILED/CORE--op--attrhand==BADPLAN-169.t         (Wstat: 0 Tests: 23 Failed: 0)
  TODO passed:   9-13, 19-23
t/C-COMPILED/CORE--op--exists_sub==BADTEST-251.t       (Wstat: 0 Tests: 23 Failed: 0)
  TODO passed:   12-13, 22-23
t/C-COMPILED/CORE--op--index==BADTEST-247.t            (Wstat: 0 Tests: 23 Failed: 0)
  TODO passed:   12-13, 22-23
t/C-COMPILED/CORE--op--packagev==BADTEST-249.t         (Wstat: 0 Tests: 23 Failed: 0)
  TODO passed:   12-13, 22-23
t/C-COMPILED/CORE--op--smartmatch==BADTEST-179.t       (Wstat: 0 Tests: 23 Failed: 0)
  TODO passed:   12-13, 22-23
t/C-COMPILED/CORE--op--ref==BADTEST-197-285-286.t      (Wstat: 0 Tests: 23 Failed: 0)
  TODO passed:   12-13, 22-23
t/C-COMPILED/CORE--op--taint==BADPLAN-288+BADTEST-288.t (Wstat: 512 Tests: 23 Failed: 2)
  Failed tests:  5, 15
  Non-zero exit status: 2
t/C-COMPILED/CORE--op--switch==BADTEST-180.t           (Wstat: 0 Tests: 23 Failed: 0)
  TODO passed:   12-13, 22-23
t/C-COMPILED/CORE--op--universal==BADTEST-299.t        (Wstat: 0 Tests: 23 Failed: 0)
  TODO passed:   12-13, 22-23
t/C-COMPILED/CORE--op--write==BADTEST-238.t            (Wstat: 0 Tests: 23 Failed: 0)
  TODO passed:   12-13, 22-23
t/C-COMPILED/CORE--re--overload==BADTEST-335.t         (Wstat: 0 Tests: 23 Failed: 0)
  TODO passed:   12-13, 22-23
t/C-COMPILED/CORE--re--pat_re_eval==SIG-332.t          (Wstat: 0 Tests: 23 Failed: 0)
  TODO passed:   8-13, 18-23
t/C-COMPILED/CORE--re--pat==BADTEST-274.t              (Wstat: 0 Tests: 23 Failed: 0)
  TODO passed:   12-13, 22-23
```

**Comment by rurban** [comment]
we still do not support the name_count < 0 case, an empty stash name.
and the name_count method seems not to work. it always return 0

**Comment by rurban** [comment]
Merged now to master with 4f91ca11037a55c773f49def2ea249023775d1a9

**Comment by atoomic** [comment]
This is still broken with HEAD @c61398f with perl522
```
> perlcc -r -e 'package Count; sub getline { print "ok\n" }; BEGIN { *The::Count:: = \*Count::; } ; The::Count->getline'
Can't locate object method "getline" via package "The::Count" (perhaps you forgot to load "The::Count"?) at -e line 1.
```

**Comment by rurban** [comment]
Fixed with e8846bbb8d6f8fd29655871791751e76d76e45a3 and a4dc466636598810f650d36cd81dc0b8c6a83f31
